app/agents_v2/primary_agent/reasoning/reasoning_engine.py

```
@@ -58,6 +59,249 @@
+                )
+            )
+            
+            # Create solution architecture
+            if primary_solution_match:
+                steps = []
+                if steps_match:
+                    # Extract bullet points or numbered items
+                    steps_text = steps_match.group(1).strip()
+                    steps = [s.strip() for s in re.split(r'[\n•\-\d\.]+', steps_text) if s.strip()]
+                
+                primary_solution = PrimarySolution(
```

Ruff (0.12.2)

Undefined name `PrimarySolution`

(F821)